main.cc

```diff
@@ -1790,9 +1792,14 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
              */
             db.local().enable_autocompaction_toggle();
 
-            sl_controller.invoke_on_all([] (qos::service_level_controller& controller) {
-                controller.set_distributed_data_accessor(::static_pointer_cast<qos::service_level_controller::service_level_distributed_data_accessor>(
-                        ::make_shared<qos::standard_service_level_distributed_data_accessor>(sys_dist_ks.local())));
+            sl_controller.invoke_on_all([&qp, &group0_client] (qos::service_level_controller& controller) -> future<> {
+                return qos::get_service_level_distributed_data_accessor_for_current_version(
+                    sys_ks.local(),
+                    sys_dist_ks.local(),
+                    qp.local(), group0_client
+                    ).then([&controller] (auto data_accessor) {
+                        controller.set_distributed_data_accessor(std::move(data_accessor));
+                    });
```

Comment:
This should probably be moved to happen after the `join_cluster` call. If the node hasn't joined the cluster yet, it doesn't know whether it joins in legacy topology mode or raft topology mode and it will unconditionally create the un-upgraded service level controller.

This is not a big issue because the node doesn't serve requests until it joins the cluster, and the SL controller will get upgraded during the join handshake, right after it does a read barrier. We may consider doing this as a followup.